test/uk/gov/hmrc/economiccrimelevyregistration/controllers/contacts/AddAnotherContactControllerSpec.scala

```diff
@@ -82,14 +84,20 @@ class AddAnotherContactControllerSpec extends SpecBase {
     "save the selected answer then redirect to the next page" in forAll {
       (registration: Registration, secondContact: Boolean) =>
         new TestContext(registration) {
-          val updatedRegistration: Registration =
-            registration.copy(contacts = registration.contacts.copy(secondContact = Some(secondContact)))
+          val updatedRegistration: Registration = secondContact match {
+            case true  => registration.copy(contacts = registration.contacts.copy(secondContact = Some(secondContact)))
+            case false =>
+              registration.copy(contacts =
+                registration.contacts
+                  .copy(secondContact = Some(secondContact), secondContactDetails = ContactDetails.empty)
+              )
+          }
```

Comment:
I think we need to avoid bringing the logic from the controller into the test like this - instead where we have 'cleanup' logic we should probably split the test out into 2 rather than interpreting what gets generated, as that's what leads us to leak logic into the tests